contributors/design-proposals/service_links.md

```diff
@@ -0,0 +1,254 @@
+# Explicit service links
+
+## Abstract
+
+Proposes an API for explicit links to services in the pod spec.
+
+## Motivation
+
+Currently, all pods are automatically injected with docker-link-style
+environment variables for each service in their namespaces.  Making service
+dependencies explicit in the PodSpec has a number of advantages:
+
+1.  Reduce accidental coupling
+2.  Reduce the number of IP tables rules that have to be created by the
+    kube-proxy
+3.  Identify relationships between pods and services, allowing user interfaces
+    to display additional information for end users.
+
+## Constraints and Assumptions
+
+1.  The old behavior must remain the default in the v1 API
+2.  It must be possible to migrate off the default to using explicit links
+
+## Use Cases
```

Comment:
there is an unspoken, but implied use case which is "my pod should not start if injected services are not available".

i would be transparent about that use case.